core/parentchain/light-client/src/concurrent_access.rs

```diff
@@ -105,10 +105,8 @@ where
 	where
 		F: FnOnce(&Self::ValidatorType) -> Result<R>,
 	{
-		let mut light_validation_lock =
+		let light_validation_lock =
 			self.light_validation.write().map_err(|_| Error::PoisonedLock)?;
-		let state = Seal::unseal_from_static_file()?;
-		light_validation_lock.set_state(state);
```

Comment:
This is a global lock; The persisted light-client file will never contain more recent information than the already initialized validator. So we can skip reading from disk.